src/XCCDF/xccdf_session.c

```diff
@@ -425,6 +425,47 @@ bool xccdf_session_set_profile_id(struct xccdf_session *session, const char *pro
 	return true;
 }
 
+bool xccdf_session_set_profile_id_by_suffix(struct xccdf_session *session, const char *profile_suffix)
+{
+	const char *full_profile_id = NULL;
+    struct xccdf_benchmark *bench = xccdf_policy_model_get_benchmark(session->xccdf.policy_model);
+
+    // Tailoring Profiles
+	struct xccdf_tailoring *tailoring = xccdf_policy_model_get_tailoring(session->xccdf.policy_model);
+	if (tailoring != NULL)   {
+		struct xccdf_profile_iterator *profit_tailoring = xccdf_tailoring_get_profiles(tailoring);
+		while (xccdf_profile_iterator_has_more(profit_tailoring)) {
+			struct xccdf_profile *tailoring_profile = xccdf_profile_iterator_next(profit_tailoring);
+			const char *tailoring_profile_id = xccdf_profile_get_id(tailoring_profile);
+
+			if(oscap_str_endswith(tailoring_profile_id, profile_suffix)) {
+				full_profile_id = tailoring_profile_id;
+				break;
+			}
+
+		}
+		xccdf_profile_iterator_free(profit_tailoring);
+	}
+
+	// Benchmark Profiles
+	if (full_profile_id != NULL) {
+		struct xccdf_profile_iterator *profit_bench = xccdf_benchmark_get_profiles(bench);
+		while (xccdf_profile_iterator_has_more(profit_bench)) {
+			struct xccdf_profile *bench_profile = xccdf_profile_iterator_next(profit_bench);
+			const char *bench_profile_id = xccdf_profile_get_id(bench_profile);
+
+			if(oscap_str_endswith(bench_profile_id, profile_suffix)) {
+				full_profile_id = bench_profile_id;
+				break;
+			}
+		}
+
+		xccdf_profile_iterator_free(profit_bench);
+	}
+
```

Comment:
`if full_profile_id == NULL` here we need to report a failure to match. passing NULL into set_profile_id means we want to select the `(default)` profile which will always be there. So we would fail silently.